### ISSUE TITLE
Deploy job passes even if some file uploads to Bintray fail

### ISSUE THREAD

**Issue by shakuzen** [micrometer-metrics/micrometer#2266]
See #2265 for example. With the last round of patch releases, I caught this happening for one of the releases, but missed it for #2265. It shouldn't rely on the person doing the release happening to see the failure in the logs or checking each and every file manually. The workflow should fail if any files fail to upload.

**Comment by shakuzen** [comment]
Apparently the applied fix did not work because the 1.5.6 release job shows as successful even though one of the Gradle runs failed. https://app.circleci.com/pipelines/github/micrometer-metrics/micrometer/855/workflows/ec03f50c-c2c2-415c-8a2b-ae7159fec457/jobs/7665

```
Caused by: org.gradle.api.GradleException: failed to upload io/micrometer/micrometer-registry-statsd/1.5.6/micrometer-registry-statsd-1.5.6.jar: HTTP 400 / {"message":"Error occurred while uploading"}
	at io.spring.gradle.bintray.task.UploadWorker.run(UploadTask.kt:107)
	... 27 more


* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.7/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 3m 54s
435 actionable tasks: 380 executed, 55 up-to-date
```

**Comment by shakuzen** [comment]
Obviated by #2455
